server/asset/btc/rpcclient.go

```diff
@@ -107,12 +108,46 @@ type txIndexResult struct {
 func (rc *RPCClient) checkTxIndex() (bool, error) {
 	res := new(txIndexResult)
 	err := rc.call(methodGetIndexInfo, anylist{"txindex"}, res)
-	if err != nil {
-		return false, err
+	if err == nil {
+		// Return early if there is no error. bitcoind returns an empty json
+		// object if txindex is not enabled. It is safe to conclude txindex is
+		// enabled if res.Txindex is not nil.
+		return res.TxIndex != nil, nil
+	}
+
+	if isMethodNotFoundErr(err) {
```

Comment:
Let's negate the condition, unindent the new code, and `return false, err` in the branch instead.